tools/cli/data_converter_plugin.go

```diff
@@ -0,0 +1,158 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cli
+
+import (
+	"fmt"
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/converter"
+)
+
+type DataConverterRPC struct {
+	client *rpc.Client
+}
+
+func NewDataConverterPlugin(name string) (converter.DataConverter, error) {
+	client, err := NewPluginClient("DataConverter", "tctl-plugin-dataconverter-"+name)
+	if err != nil {
+		return nil, fmt.Errorf("unable to register plugin: %v\n", err)
+	}
+
+	dataConverter, ok := client.(converter.DataConverter)
+	if !ok {
+		return nil, fmt.Errorf("incorrect plugin type")
+	}
+
+	return dataConverter, nil
+}
+
+func (g *DataConverterRPC) FromPayload(payload *commonpb.Payload, valuePtr interface{}) error {
+	err := g.client.Call("Plugin.FromPayload", payload, valuePtr)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (g *DataConverterRPC) FromPayloads(payloads *commonpb.Payloads, valuePtr ...interface{}) error {
+	err := g.client.Call("Plugin.FromPayloads", payloads, valuePtr)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (g *DataConverterRPC) ToPayload(value interface{}) (*commonpb.Payload, error) {
+	var payload commonpb.Payload
+	err := g.client.Call("Plugin.ToPayload", value, &payload)
+	if err != nil {
+		return nil, err
+	}
+
+	return &payload, nil
+}
+
+func (g *DataConverterRPC) ToPayloads(values ...interface{}) (*commonpb.Payloads, error) {
+	var payloads commonpb.Payloads
+	err := g.client.Call("Plugin.ToPayloads", values, &payloads)
+	if err != nil {
+		return nil, err
+	}
+
+	return &payloads, nil
+}
+
+func (g *DataConverterRPC) ToString(input *commonpb.Payload) string {
+	var resp string
+	err := g.client.Call("Plugin.ToString", input, &resp)
+	if err != nil {
+		return err.Error()
+	}
+
+	return resp
+}
+
+func (g *DataConverterRPC) ToStrings(input *commonpb.Payloads) []string {
+	var resp []string
+	err := g.client.Call("Plugin.ToStrings", input, &resp)
+	if err != nil {
+		return []string{err.Error()}
+	}
+
+	return resp
+}
+
+type DataConverterRPCServer struct {
+	Impl converter.DataConverter
```

Comment:
Is it some sort of convension? I can't find where this field is initialized.

Reply:
Yeah it's what the examples use: https://github.com/hashicorp/go-plugin/blob/master/examples/basic/commons/greeter_interface.go

It's initialised here: https://github.com/temporalio/temporal/pull/1437/files#diff-84f6aa99b0e210c9aa282f264ebab96f55dd30987bfbcb7c04e4b5c5c6566642R153

We can use a different name if we want to.